app/models/sitemap.rb

```diff
@@ -0,0 +1,14 @@
+class Sitemap < ActiveRecord::Base
+  include Fetchable
+
+  attr_readonly :url
+
+  belongs_to :searchgov_domain
+
+  validates_associated :searchgov_domain, on: :create
+  validates_presence_of :searchgov_domain, on: :create
+
+  before_validation :set_searchgov_domain, on: :create
```

Comment:
Sorry, I didn't catch this earlier. Because this is a `before_validation` callback, it makes logical sense to place this line before the validation lines in the code.